presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -830,4 +832,16 @@ public static void closeWithSuppression(RecordCursor recordCursor, Throwable thr
             }
         }
     }
+
+    public static List<HiveType> getAllStructFieldTypeInfos(HiveType hiveType)
```

Comment:
`getAllStructFieldTypeInfos` is a particularly bad name because it's the same as the one you are using 2 lines below, but it's something quite different.

I recommend inlining this function and the one below.

Reply:
It's referred to multiple places like the coercers for structs. I agree that the name is confusing. I would rename it as `getAllStructFieldHiveTypes`.